www/nginx/src/opnsense/scripts/nginx/ngx_functions.js

```diff
@@ -0,0 +1,60 @@
+var fs = require('fs');
+var tls_fingerprints = JSON.parse(fs.readFileSync('tls_fingerprints.json'));
+
+function customArrayIndexOf(haystack, needle) {
+  var element, element_id;
+  for (element_id in haystack) {
+    element = haystack[element_id];
+    if (element == needle) return element_id;
+  }
+  return -1;
+}
```

Comment:
sorry, this function is needed, Array.indexOf does not work for some reasons (don't know why)…

Reply:
ok, just for reference: which browser in particular?

Reply:
this is not a browser, this runs directly inside nginx (njs). It is strange that when I use the njs comand line util it works as expected but in nginx it is strange:
```js
var a = 'x';
var b = 'x':
a == b // true
a === b // false but should be true - indexOf probably uses this comparison
```

Reply:
I think the problem is fixed by this commit: https://github.com/nginx/njs/commit/f0dc7105c37f656cb2c5ce073998420e98eff3b4
Could you build njs from master branch and test?

Reply:
@VBart I just tried indexOf on version nginx/njs@549e51ec1db394fb7bdccb8e21026e912a19c012 which I have built yesterday. It still returns `-1`

nginx version: Nginx-1.14.0 (if it helps)

Reply:
@fabianfrz It doesn't include the fix I mentioned. The fix was committed 16 days ago (and will be released with the upcoming 0.2.7), while the version you've tried was released about half a year ago.

Reply:
@VBart tested against nginx/njs@36f12955c906d35282c32222b99c7861a6f805f2 and it works. Do you already know a release date?

BTW: thanks for spending your time on this.

Reply:
@fabianfrz There's no specific date set yet. But according to usual release frequency and taking into account New Year holidays, most likely it will happen in the middle of January.

Thank you for testing. :beers:

Reply:
Btw, since the problem is related to comparsion of UTF-8 and Byte strings internal representation, I think you can try to workaround it by using `toBytes()` or `toUTF8()` methods. Not sure which one will work in this particular case:

```javascript
current_index = fingerprint_ciphers.indexOf(browser_cipher.toUTF8());
```
or

```javascript
current_index = fingerprint_ciphers.indexOf(browser_cipher.toBytes());
```

Reply:
@fabianfrz jfyi, njs 0.2.7 has been released today with the fix.

As of FreeBSD build issue, that's strange, because njs works perfectly on FreeBSD. And njs module is a part of official nginx port for a long time: https://www.freshports.org/www/nginx

Reply:
@VBart thanks, so let's wait for inclusion in the FreeBSD ports tree so we can use it. @fichtner fixed the build by patching out a build flag:

https://github.com/opnsense/ports/commit/d188643649f7af027d4f955f0497cd5a4cae42a4#diff-60d367a88ba4945c90d6504cc2f25228

It seems that an unsupported compiler flag caused the issue.